routes/transaction.go

```diff
@@ -2654,66 +2637,196 @@ func (fes *APIServer) CreateDAOCoinLimitOrder(ww http.ResponseWriter, req *http.
 		)
-		if err != nil {
-			_AddBadRequestError(
-				ww,
-				fmt.Sprintf(
-					"CreateDAOCoinLimitOrder: Error getting public key for %v: %v",
-					requestData.SellingDAOCoinCreatorPublicKeyBase58CheckOrUsername,
-					err,
-				),
-			)
-		}
+		return
+	}
+
+	utxoView, err := fes.backendServer.GetMempool().GetAugmentedUniversalView()
+	if err != nil {
+		_AddInternalServerError(ww, fmt.Sprintf("CreateDAOCoinMarketOrder: problem fetching utxoView: %v", err))
+		return
 	}
 
+	// Decode and validate the buying / selling coin public keys
+	buyingCoinPublicKey, sellingCoinPublicKey, err := fes.getBuyingAndSellingDAOCoinPublicKeys(
+		utxoView,
+		requestData.BuyingDAOCoinCreatorPublicKeyBase58CheckOrUsername,
+		requestData.SellingDAOCoinCreatorPublicKeyBase58CheckOrUsername,
+	)
+	if err != nil {
+		_AddBadRequestError(ww, fmt.Sprintf("CreateDAOCoinMarketOrder: %v", err))
+		return
+	}
+
+	// override the initial value and explicitly set to 0 for clarity
+	zeroUint256 := uint256.NewInt().SetUint64(0)
+
 	res, err := fes.createDAOCoinLimitOrderResponse(
 		utxoView,
 		requestData.TransactorPublicKeyBase58Check,
 		buyingCoinPublicKey,
 		sellingCoinPublicKey,
-		scaledExchangeRateCoinsToSellPerCoinToBuy,
+		zeroUint256,
 		quantityToFillInBaseUnits,
 		operationType,
+		fillType,
 		nil,
 		requestData.MinFeeRateNanosPerKB,
 		requestData.TransactionFees,
 	)
 	if err != nil {
-		_AddInternalServerError(ww, fmt.Sprintf("CreateDAOCoinLimitOrder: %v", err))
+		_AddInternalServerError(ww, fmt.Sprintf("CreateDAOCoinMarketOrder: %v", err))
 		return
 	}
 
 	if err = json.NewEncoder(ww).Encode(res); err != nil {
-		_AddInternalServerError(ww, fmt.Sprintf("CreateDAOCoinLimitOrder: Problem encoding response as JSON: %v", err))
+		_AddInternalServerError(ww, fmt.Sprintf("CreateDAOCoinMarketOrder: Problem encoding response as JSON: %v", err))
 		return
 	}
 }
 
+// getBuyingAndSellingDAOCoinPublicKeys
+// An empty string for the buying or selling coin represents $DESO. This enables $DESO <> DAO coin trades, and
+// DAO coin <> DAO coin trades. At most one of the buying or selling coin can specify $DESO as we don't enable
+// $DESO <> $DESO trades
+func (fes *APIServer) getBuyingAndSellingDAOCoinPublicKeys(
+	utxoView *lib.UtxoView,
+	buyingDAOCoinCreatorPublicKeyBase58CheckOrUsername string,
+	sellingDAOCoinCreatorPublicKeyBase58CheckOrUsername string,
+) (_buyingCoinPublicKey []byte, _sellingCoinPublicKey []byte, _err error) {
+	if sellingDAOCoinCreatorPublicKeyBase58CheckOrUsername == "" &&
+		buyingDAOCoinCreatorPublicKeyBase58CheckOrUsername == "" {
+		return nil, nil, errors.Errorf("empty string provided for both the " +
+			"coin to buy and the coin to sell. At least one must specify a valid DAO public key or username whose coin " +
+			"will be bought or sold")
+	}
+
+	_buyingCoinPublicKey = lib.ZeroPublicKey.ToBytes()
+	_sellingCoinPublicKey = lib.ZeroPublicKey.ToBytes()
```

Comment:
```suggestion
	buyingCoinPublicKey := lib.ZeroPublicKey.ToBytes()
	sellingCoinPublicKey := lib.ZeroPublicKey.ToBytes()
```
Let's avoid using the name returned values feature. Diamondhands isn't a fan of these as they can be abused to make code harder to read. I don't think that's the case here, but let's stick with distinct names from the return values.